src/v8/inspector/mod.rs

```diff
@@ -0,0 +1,425 @@
+    /// called by this object, but by the C++ side.
+    _on_wait_frontend_message_on_pause_callback: Box<Box<OnWaitFrontendMessageOnPauseCallback>>,
+}
+
+impl std::fmt::Debug for Inspector {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let on_response_str = {
+            let callback = &self._on_response_callback;
+            format!("Some({callback:p})")
+        };
+
+        let on_wait_str = {
+            let callback = &self._on_wait_frontend_message_on_pause_callback;
+            format!("Some({callback:p})")
+        };
+
+        f.debug_struct("Inspector")
+            .field("raw", &self.raw)
+            .field("_on_response_callback", &on_response_str)
+            .field("_on_wait_frontend_message_on_pause_callback", &on_wait_str)
+            .finish()
+    }
+}
+
+/// The callback function used to send the messages back to the client
+/// of the inspector (to the one who is debugging).
+extern "C" fn on_response(
+    string: *const ::std::os::raw::c_char,
+    rust_callback: *mut ::std::os::raw::c_void,
+) {
+    let string = unsafe { std::ffi::CStr::from_ptr(string) }.to_string_lossy();
+    log::trace!("Outgoing message: {string}");
+    let rust_callback: &mut Box<OnResponseCallback> = unsafe {
+        &mut *(rust_callback as *mut std::boxed::Box<dyn std::ops::FnMut(std::string::String)>)
+    };
+    rust_callback(string.to_string())
+}
+
+extern "C" fn on_wait_frontend_message_on_pause(
+    raw: *mut crate::v8_c_raw::bindings::v8_inspector_c_wrapper,
+    rust_callback: *mut ::std::os::raw::c_void,
+) -> ::std::os::raw::c_int {
+    log::trace!("on_wait_frontend_message_on_pause");
+    let rust_callback: &mut Box<OnWaitFrontendMessageOnPauseCallback> = unsafe {
+        &mut *(rust_callback
+            as *mut std::boxed::Box<
+                dyn std::ops::FnMut(*mut crate::v8_c_raw::bindings::v8_inspector_c_wrapper) -> i32,
+            >)
+    };
+    rust_callback(raw)
+}
+
+impl Inspector {
+    /// Creates a new [Inspector].
+    pub fn new(
+        raw: Arc<RawInspector>,
+        on_response_callback: Box<OnResponseCallback>,
+        on_wait_frontend_message_on_pause_callback: Box<OnWaitFrontendMessageOnPauseCallback>,
+    ) -> Self {
+        let on_response_callback = Self::set_on_response_callback(&raw, on_response_callback);
+        let on_wait_callback = Self::set_on_wait_frontend_message_on_pause_callback(
+            &raw,
+            on_wait_frontend_message_on_pause_callback,
+        );
+        Self {
+            raw,
+            _on_response_callback: on_response_callback,
+            _on_wait_frontend_message_on_pause_callback: on_wait_callback,
+        }
+    }
+
+    /// Sets the callback which is used by the debugger to send messages
+    /// to the remote client.
+    fn set_on_response_callback(
+        raw: &RawInspector,
+        on_response_callback: Box<OnResponseCallback>,
```

Comment:
Why not using closures here?

Reply:
Ohh is it a closure, but I wonder why not to use generics and hide th Box thing to be an implementation detail.

Reply:
But why you need to use `dyn` here, please look at the following example: https://github.com/RedisGears/v8-rs/blob/139b25e662331515615aff97440e97234037fd29/src/v8/v8_native_function_template.rs#L36
The C function can also be generic.